main.go

```diff
@@ -173,52 +230,97 @@ func getProjectID(baseURL string, token string, projectname string, namespace st
 }
 
 // Get and return currently active milestones
-func getActiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
-	state := "active"
-	return getMilestones(baseURL, token, projectID, state)
+func getActiveMilestones(baseURL string, token string, projectID string, api string) ([]milestoneAPI, error) {
+	var state string
+	switch {
+	case api == "gitlab":
+		state = "active"
+	case api == "github":
+		state = "open"
+	}
+	return getMilestones(baseURL, token, projectID, state, api)
 }
 
 // Get and return inactive milestones
-func getInactiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
+func getInactiveMilestones(baseURL string, token string, project string, api string) ([]milestoneAPI, error) {
 	state := "closed"
-	return getMilestones(baseURL, token, projectID, state)
+	return getMilestones(baseURL, token, project, state, api)
 }
 
-func reactivateClosedMilestones(milestones map[string]milestone, baseURL string, token string, projectID string) error {
+func reactivateClosedMilestones(milestones map[string]milestone, baseURL string, token string, project string, api string) error {
 	client := &http.Client{}
+	var strURL []string
 	for _, v := range milestones {
-		milestoneID := v.ID
-		strURL := []string{baseURL, "/projects/", projectID, "/milestones/", milestoneID}
-		URL := strings.Join(strURL, "")
-
-		// Overwrite state information in URL
-		u, _ := url.Parse(URL)
-		q := u.Query()
-		q.Set("state_event", "activate")
-		u.RawQuery = q.Encode()
-		req, err := http.NewRequest("PUT", u.String(), nil)
-		if err != nil {
-			logger.Println(err)
+		switch {
+		case api == "gitlab":
+			milestoneID := v.ID
+			strURL = []string{baseURL, "/projects/", project, "/milestones/", milestoneID}
+		case api == "github":
+			milestoneID := strconv.Itoa(v.Number)
+			strURL = []string{baseURL, project, "/milestones/", milestoneID}
 		}
-		req.Header.Add("PRIVATE-TOKEN", token)
-		resp, err := client.Do(req)
-		if err != nil {
-			return err
+		URL := strings.Join(strURL, "")
+		switch {
+		case api == "gitlab":
+			// Overwrite state information in URL
+			u, _ := url.Parse(URL)
+			q := u.Query()
+			q.Set("state_event", "activate")
+			u.RawQuery = q.Encode()
+			req, err := http.NewRequest("PUT", u.String(), nil)
+			if err != nil {
+				logger.Println(err)
+			}
+			req.Header.Add("PRIVATE-TOKEN", token)
+			resp, err := client.Do(req)
+			if err != nil {
+				return err
+			}
+			defer resp.Body.Close()
+		case api == "github":
+			create := struct {
```

Comment:
It's not a create activity, but an update one. Might be clearer to say "updatePatch"/"updateBlob" or so.

Reply:
Also don't like the inline struct here, but let's go with this for now.

Reply:
Thanks, I changed the naming to better reflect the activity.